src/js/components/DataTable/Body.js

```diff
@@ -85,67 +89,96 @@ const Body = forwardRef(
                 ? datumValue(datum, primaryProperty)
                 : undefined;
               const isSelected = selected && selected.includes(primaryValue);
+              const isrowExpanded = rowExpand && rowExpand.includes(index);
               return (
-                <StyledDataTableRow
-                  key={primaryValue || index}
-                  ref={rowRef}
-                  size={size}
-                  active={active >= 0 ? active === index : undefined}
-                  onClick={
-                    onClickRow
-                      ? event => {
-                          // extract from React's synthetic event pool
-                          event.persist();
-                          const adjustedEvent = event;
-                          adjustedEvent.datum = datum;
-                          adjustedEvent.index = index;
-                          onClickRow(adjustedEvent);
-                        }
-                      : undefined
-                  }
-                  onMouseEnter={onClickRow ? () => setActive(index) : undefined}
-                  onMouseLeave={
-                    onClickRow ? () => setActive(undefined) : undefined
-                  }
-                  onFocus={onClickRow ? () => setActive(index) : undefined}
-                  onBlur={onClickRow ? () => setActive(undefined) : undefined}
-                >
-                  {(selected || onSelect) && (
-                    <TableCell background={background}>
-                      <CheckBox
-                        a11yTitle={`${
-                          isSelected ? 'unselect' : 'select'
-                        } ${primaryValue}`}
-                        checked={isSelected}
-                        disabled={!onSelect}
-                        onChange={() => {
-                          if (isSelected)
-                            onSelect(selected.filter(s => s !== primaryValue));
-                          else onSelect([...selected, primaryValue]);
+                <Fragment key={`${index.toString()}_expanded`}>
```

Comment:
I think we can move the key from line 96 to here and omit the key at line 96 below.